src/lib/MedStickerEncryptionCharlie.js

```diff
@@ -0,0 +1,80 @@
+import create from "./factory";
+import { arrayBufferToHex } from "./utilities";
+import { CHARLIE } from "./constants";
+
+const scrypt = create("scrypt");
+const gcm = create("AES-GCM");
+
+/**
+ * @param secret {string}
+ * @param salt {string}
+ * @returns {Array}
+ */
+export function hash(secret, salt) {
+  try {
+    return scrypt.generateKey(secret, salt);
+  } catch (e) {
+    throw new Error("HashFailed");
+  }
+}
+
+/**
+ * Given a hashed string, returns a concatenated string
+ * that contains the version of the encryption its being
+ * used and the incoming string in hex format
+ * @param hashed {string}
+ * @returns {string}
+ */
+
+export function fingerprint(hashed) {
+  const hexFingerprintSecret = arrayBufferToHex(hashed);
+  return `${CHARLIE}:${hexFingerprintSecret}`;
+}
+
+/**
+ * Takes an array, splits it in half,
+ * from the first bit creates an ArrayBuffer
+ * from the second bit creates a string that
+ * contains a secret but also indicates the version of the algorithm used
+ * @param array {Array}
+ * @returns {{fingerprintFile: string, key: ArrayBufferLike}}
+ */
+export function splitKeys(array) {
+  const arrayLength = array.length;
+  const key = array.slice(0, arrayLength / 2);
+  const fingerprintFile = array.slice(arrayLength / 2, arrayLength);
+  const fingerprintArrayBuffer = new Uint8Array(fingerprintFile).buffer;
```

Comment:
please inline this variable, 

btw isn't that already array buffer after splitting an array buffer?

Reply:
the incoming value is array. on my next push this will be clearer.